pkg/mtu/mtu.go

```diff
@@ -110,11 +116,12 @@ func NewConfiguration(authKeySize int, encryptEnabled bool, encapEnabled bool, m
 	}
 
 	conf := Configuration{
-		standardMTU:    mtu,
-		tunnelMTU:      mtu - (TunnelOverhead + encryptOverhead),
-		encryptMTU:     mtu - encryptOverhead,
-		encapEnabled:   encapEnabled,
-		encryptEnabled: encryptEnabled,
+		standardMTU:      mtu,
+		tunnelMTU:        mtu - (TunnelOverhead + encryptOverhead),
+		encryptTunnelMTU: mtu - TunnelOverhead,
```

Comment:
The above two were counter-intuitive to me at first glance, because the `encryptOverhead` is not added to the variable with `encrypt` in the name. I think that the initialization here is consistent with the comments for the variables above, but it is suprising at a glance. I wonder if there's another name we can use to make this more clear?

Reply:
open for suggestions... tunnelMTUAfterEncrypt?

Reply:
`preEncryptTunnelMTU` / `postEncryptTunnelMTU`?

Reply:
I dropped the mtu test for the moment as I think up something better hard coding the mtu breaks various test environments.

Reply:
keeping the wip because I want to test in a few clusters with different mtus before it gets merged.

Reply:
@joestringer landed on preEncryptMTU and postEncryptMTU because we use those values in the direct routing case as well and it was a bit odd to use preEncryptTunnelMTU directly after a check !EncapEnabled.